File: Server Integration/PHP/RTCHOOKS.php
<?php

class RTCHOOKS {

	function RTCHOOKS() {
		
		$this->methodTable = array (
			"receiveNode" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			),
			"receiveNodeDeletion" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			),
			"receiveItem" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			),
			"receiveItemRetraction" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			),
			"receiveNodeConfiguration" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			),
			"receiveItems" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			),
			"receiveNodes" => array (
				"access" => "remote",
				"description" => "Pings back a message"
			)
		);
	}


	/**
	 * String securityToken,
	  String roomName,
	  String collectionName,
	  String nodeName
	  Map config
	 */
	function receiveNode($token, $roomName, $collectionName, $nodeName, $config) {
		$binary = " recieveNode: " . $roomName . " " . $collectionName . " " . $nodeName . "\n";
		$binary .= print_r($config, true);
		//return $this->writeToFile($binary);
	}

	/**
	 * String securityToken,
	  String roomName,
	  String collectionName,
	  String nodeName
	 */
	function receiveNodeDeletion($token, $roomName, $collectionName, $nodeName) {
		$binary = " receiveNodeDeletion: " . $roomName . " " . $collectionName . " " . $nodeName . "\n";
		//return $this->writeToFile($binary);
	}

	/**
	 * String securityToken,
	  String roomName,
	  String collectionName,
	  Object messageItemVO
	 */
	function receiveItem($token, $roomName, $collectionName, $itemObj) {
		$binary = " receiveItem: " . $roomName . " " . $collectionName . "\n";
		$binary .= print_r($itemObj, true);
		//$this->writeToFile($binary . " retrieving Filter \n");
	}

	/*
	 * String securityToken,
	  String roomName,
	  String collectionName,
	  String nodeName,
	  Object messageItemVO
	 */
	function receiveItemRetraction($token, $roomName, $collectionName, $nodeName, $itemObj) {
		$binary = " receiveItemRetraction: " . $roomName . " " . $collectionName . " " . $nodeName . "\n";
		$binary .= print_r($itemObj, true);
		//return $this->writeToFile($binary);
	}

	/*
	 * String securityToken,
	  String roomName,
	  String collectionName,
	  String nodeName,
	  Object nodeconfig
	 */
	function receiveNodeConfiguration($token, $roomName, $collectionName, $nodeName, $config) {
		$binary = " receiveNodeConfiguration: " . $roomName . " " . $collectionName . " " . $nodeName . "\n";
		$binary .= print_r($config, true);
		//return $this->writeToFile($binary);
	}

}
?>

File: Server Integration/CSharp/RTCHOOKS.cs
using System;
using System.Collections.Generic;
using System.Text;

//if you using FluorineFx, make sure include the package and indicae remotingservice
//using FluorineFx; [RemotingService("RTCHOOKS")]

public class RTCHOOKS
{
    public void receiveNode(string token, string roomName, string collectionName, string nodeName, Dictionary<string, object> config)
    {
        Console.WriteLine("ReceiveNode: " + token + " " + roomName + " " + collectionName + " " + nodeName);
    }

    public void receiveNodeDeletion(string token, string roomName, string collectionName, string nodeName) 
    {
        Console.WriteLine("receiveNodeDeletion: " + token + " " + roomName + " " + collectionName + " " + nodeName);
    }

    public void receiveItem(string token, string roomName, string collectionName, Dictionary<string, object> itemObj) 
    {
        Console.WriteLine("receiveItem: " + token + " " + roomName + " " + collectionName);
    }

    public void receiveItemRetraction(string token, string roomName, string collectionName, string nodeName, Dictionary<string, object> itemObj)
    {
        Console.WriteLine("receiveItemRetraction: " + token + " " + roomName + " " + collectionName + " " + nodeName);
    }

    public void receiveNodeConfiguration(string token, string roomName, string collectionName, string nodeName, Dictionary<string, object> config) 
    {
        Console.WriteLine("receiveNodeConfiguration: " + token + " " + roomName + " " + collectionName + " " + nodeName);
    }

    public void receiveUserRole(string token, string roomName, string collectionName, string nodeName, string userID, int role) 
    {
        Console.WriteLine("receiveUserRole: " + token + " " + roomName + " " + collectionName + " " + nodeName + " " + userID + " " + role);
    }
}

